storage/rocksdb/ha_rocksdb.cc

```diff
@@ -9705,18 +9705,36 @@ void ha_rocksdb::setup_iterator_bounds(
     const Rdb_key_def &kd, const rocksdb::Slice &eq_cond, size_t bound_len,
     uchar *const lower_bound, uchar *const upper_bound,
     rocksdb::Slice *lower_bound_slice, rocksdb::Slice *upper_bound_slice) {
-  uint min_len = std::min(eq_cond.size(), bound_len);
-  memcpy(upper_bound, eq_cond.data(), min_len);
-  kd.successor(upper_bound, min_len);
-  memcpy(lower_bound, eq_cond.data(), min_len);
-  kd.predecessor(lower_bound, min_len);
+  // bound_len represents the lower_bound and upper_bound buffer size. It
+  // should always be greater than Rdb_key_def::INDEX_NUMBER_SIZE because
+  // callers either with a local buffer of size Rdb_key_def::INDEX_NUMBER_SIZE
+  // or it is called with m_scan_it_lower_bound.
+  DBUG_ASSERT(bound_len >= Rdb_key_def::INDEX_NUMBER_SIZE);
```

Comment:
We can probably move this DBUG_ASSERT into the first conditional and assert it's Rdb_key_def::INDEX_NUMBER_SIZE.